torch_xla/distributed/xla_dist.py

```diff
@@ -86,14 +96,38 @@ def __init__(self,
                ' will interfere with the values set for distributed'
                ' training'.format(dist_var)))
 
+  def _check_client_mesh_health(self):
+
+    def _pop_mark_step_queue(queue, timeout=30):
+      queue.get(timeout=timeout)
+
+    with futures.ThreadPoolExecutor(
+        max_workers=len(self._heartbeats)) as executor:
+      try:
+        results = executor.map(_pop_mark_step_queue, self._heartbeats.values())
+        for result in results:
+          if result:
+            result.result()  # Only iterating to re-raise any worker errors.
+      except queue.Empty:
+        self._error_queue.put(RuntimeError(
+          'Client mesh is unhealthy due to non-matching heartbeats'))
+
   def _stream_logs(self, process, client_worker):
     client_ip = client_worker._internal_ip
```

Comment:
More internal members access.
We better expose this via APIs.

Reply:
I've changed `_internal_ip` to be a public member (`internal_ip`) instead. Do you think we should rather have getters?